src/main/java/org/spongepowered/api/service/profile/GameProfileResolver.java

```diff
@@ -0,0 +1,135 @@
+
+/**
+ * A service which contacts the authentication servers to get a
+ * {@link GameProfile} by a given UUID or name.
+ *
+ * <p>The service may cache the data of a request for faster lookups. Note that
+ * the cached data may not always be up to date.</p>
+ *
+ * <p>The returned {@link ListenableFuture} throws an {@link ExecutionException}
+ * caused by a {@link ProfileNotFoundException} if the profile does not exist or
+ * an {@link IOException} if a network error occurred.</p>
+ */
+public interface GameProfileResolver {
+
+    /**
+     * Looks up a {@link GameProfile} by its unique ID.
+     *
+     * <p>This method checks the local profile cache before contacting the
+     * profile servers. Use {@link #get(UUID, boolean)} to disable the cache
+     * lookup.</p>
+     *
+     * @param uniqueId The unique ID
+     * @return The result of the request
+     */
+    ListenableFuture<GameProfile> get(UUID uniqueId);
+
+    /**
+     * Looks up a {@link GameProfile} by its unique ID.
+     *
+     * @param uniqueId The unique ID
+     * @param useCache true to perform a cache lookup first
+     * @return The result of the request
+     */
+    ListenableFuture<GameProfile> get(UUID uniqueId, boolean useCache);
+
+    /**
+     * Looks up a {@link GameProfile} by its user name (case-insensitive).
+     *
+     * <p>This method checks the local profile cache before contacting the
+     * profile servers. Use {@link #get(String, boolean)} to disable the cache
+     * lookup.</p>
+     *
+     * @param name The username
+     * @return The result of the request
+     */
+    ListenableFuture<GameProfile> get(String name);
+
+    /**
+     * Looks up a {@link GameProfile} by its user name (case-insensitive).
+     *
+     * @param name The username
+     * @param useCache true to perform a cache lookup first
+     * @return The result of the request
+     */
+    ListenableFuture<GameProfile> get(String name, boolean useCache);
+
+    /**
+     * Gets a collection of {@link GameProfile}s by their user names
+     * (case-insensitive).
+     *
+     * @param names The user names
+     * @param useCache true to perform a cache lookup first
+     * @return The result of the request
+     */
+    ListenableFuture<Collection<GameProfile>> getAllByName(Iterable<String> names, boolean useCache);
+
+    /**
+     * Gets a collection of {@link GameProfile}s by their unique IDs.
+     *
+     * @param uniqueIds The UUIDs
+     * @param useCache true to perform a cache lookup first
+     * @return The result of the request
+     */
+    ListenableFuture<Collection<GameProfile>> getAllById(Iterable<UUID> uniqueIds, boolean useCache);
+
+    /**
+     * Returns a collection of matching cached {@link GameProfile}s whose last
+     * known user names start with the given string (case-insensitive).
+     *
+     * <p>This collection may also contain profiles of players who never played
+     * on the server!</p>
+     *
```

Comment:
If this only fetches from the local cache then how can it contain profiles of players who have never played?